internal/dataplane/translator/subtranslator/httproute.go

```diff
@@ -544,43 +617,130 @@ func kongHeaderFormatter(header gatewayapi.HTTPHeader) string {
 			}
+			return plugin, nil, nil
 
-			plugin := kong.Plugin{
-				Name: kong.String("request-transformer"),
-				Config: kong.Configuration{
-					"replace": map[string]string{
-						"uri": *filter.Path.ReplaceFullPath,
-					},
-				},
+		case gatewayapi.PrefixMatchHTTPPathModifier:
+			if expressionsRouterEnabled {
+				// TODO: https://github.com/Kong/kubernetes-ingress-controller/issues/3686
+				return kong.Plugin{}, nil, fmt.Errorf("%s unsupported with expressions router", gatewayapi.PrefixMatchHTTPPathModifier)
 			}
 
-			return plugin, nil
-
-		// TODO: https://github.com/Kong/kubernetes-ingress-controller/issues/3686
-		case gatewayapi.PrefixMatchHTTPPathModifier:
-			return kong.Plugin{}, fmt.Errorf("%s unsupported for %s", gatewayapi.PrefixMatchHTTPPathModifier, gatewayapi.HTTPRouteFilterURLRewrite)
+			plugin, routeModifier := generateRequestTransformerForURLRewritePrefixMatch(filter, path)
+			return plugin, routeModifier, nil
 		}
 	}
 
 	// TODO: https://github.com/Kong/kubernetes-ingress-controller/issues/3685
 	if filter.Hostname != nil {
-		return kong.Plugin{}, fmt.Errorf("unsupported hostname replace for %s", gatewayapi.HTTPRouteFilterURLRewrite)
+		return kong.Plugin{}, nil, fmt.Errorf("unsupported hostname replace for %s", gatewayapi.HTTPRouteFilterURLRewrite)
+	}
+
+	return kong.Plugin{}, nil, fmt.Errorf("invalid %s config", gatewayapi.HTTPRouteFilterURLRewrite)
+}
+
+// generateRequestTransformerForURLRewriteFullPath generates a request-transformer plugin for the URLRewrite filter
+// with a FullPathHTTPPathModifier.
+func generateRequestTransformerForURLRewriteFullPath(filter *gatewayapi.HTTPURLRewriteFilter) (kong.Plugin, error) {
+	if filter.Path.ReplaceFullPath == nil {
+		return kong.Plugin{}, fmt.Errorf("%s missing ReplaceFullPath", gatewayapi.HTTPRouteFilterURLRewrite)
+	}
+
+	return kong.Plugin{
+		Name: kong.String("request-transformer"),
+		Config: kong.Configuration{
+			"replace": map[string]string{
+				"uri": *filter.Path.ReplaceFullPath,
+			},
+		},
+	}, nil
+}
+
+// generateRequestTransformerForURLRewritePrefixMatch generates a request-transformer plugin and a route modifier
+// for the URLRewrite filter with a PrefixMatchHTTPPathModifier.
+func generateRequestTransformerForURLRewritePrefixMatch(filter *gatewayapi.HTTPURLRewriteFilter, path string) (kong.Plugin, kongRouteModifier) {
+	// By default, we replace the URI with a slash.
+	replaceURI := "/"
+
+	// If an empty path is provided, we need to make sure that the path will always start with a slash.
+	if path == "" {
+		path = "/"
+	}
+
+	// If the ReplacePrefixMatch is provided, we need to use it as the replacement URI.
+	if replacePrefixMatch := filter.Path.ReplacePrefixMatch; replacePrefixMatch != nil {
+		// Trim the trailing slash from the ReplacePrefixMatch to avoid double slashes in the final URI.
+		replaceURI = strings.TrimSuffix(*replacePrefixMatch, "/")
```

Comment:
How about extracting the logic of calculating replace URI to another function?

Reply:
Split `generateRequestTransformerForURLRewritePrefixMatch` to a function responsible for generating the request transformer's URI replace string, and another one for generating a Kong Route modifier: https://github.com/Kong/kubernetes-ingress-controller/pull/5895/commits/7c6e3a08e39168dcf119fc948ba199c9dde41f95